### ISSUE TITLE
feat(oxc_semantic): jsdoc for other AstTypes

### ISSUE THREAD

**Issue by etkachev** [oxc-project/oxc#1506]
Hello! Was wondering if the jsdoc builder can also register jsdocs for docs that are included on properties and methods on classes for example.

Looking through the [code](https://github.com/oxc-project/oxc/blob/main/crates/oxc_semantic/src/jsdoc/builder.rs#L26), it looks like it only registers those that are declarations only. Would be nice to be able to extract jsdoc for things like this:

```typescript
@Component({
   selector: 'app-card-section',
})
export class CardSectionComponent {
  /**
   * text to display in top header
   */
  @Input() headerText: string;
  /**
   * secondary text
   */
  @Input() subText: string;
  /**
   * whether to show divider between sections
   */
  @Input() showDivider = true;

  /**
   * expand the card accordion section
   */
  expand() {
    // TODO
  }
}
```

**Comment by shannonrothe** [comment]
These would be considered decorators not jsdoc annotations right @Boshen?

**Comment by Boshen** [comment]
The current jsdoc build is incomplete, we welcome PRs that can expand its functionalities.

@etkachev Are you looking into oxc for building something new in Rust?

**Comment by Boshen** [comment]
> These would be considered decorators not jsdoc annotations right @Boshen?

I don't get your questions 🤔 

But I looked up comment attachment in AST explorer and have confirmed that

`@Input() headerText: string;` should have the jsdoc comment `/**\n* text to display in top header\n*/`

**Comment by shannonrothe** [comment]
Ah sorry, I thought they were referring to parsing the decorators on those class fields as jsdoc annotations!

I can take a look at handling more cases with jsdoc comments soon 👍🏻

**Comment by etkachev** [comment]
@Boshen @shannonrothe  I saw that you guys merged some updates to the crates. Is there a timeline on how often you guys push out updates for these?

**Comment by Boshen** [comment]
I've published the latest changes as v0.4.0 to crates.io
